### ISSUE TITLE
API Authentication Issue

### ISSUE THREAD

**Issue by terra-nova** [Icinga/icingaweb2-module-director#1343]
I'd like to try out the Director REST API but no matter what, I always get an 401 Unauthorized HTTP error.

The Icingaweb2 user account I tested with has all permssions (`[ "*" ]`).

I am following the instructions from here: https://github.com/Icinga/icingaweb2-module-director/blob/master/doc/70-REST-API.md

I added -vvv to the curl call in the director-curl script in the hope of spotting some error, however I fail to see anything suspicious other than the fact that the server returns HTTP status code 401.

```
$ ./director-curl GET 'director/host?name=testhost'
* About to connect() to icinga-test port 443 (#0)
*   Trying 192.168.1.1...
* Connected to icinga-test (192.168.1.1) port 443 (#0)
* Initializing NSS with certpath: sql:/etc/pki/nssdb
*   CAfile: /etc/pki/tls/certs/ca-bundle.crt
  CApath: none
* SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate:
*       ...
* Server auth using Basic with user 'admin'
> GET /icingaweb2/director/host?name=testhost HTTP/1.1
> Authorization: Basic YWRtaW46YWRtaW4=
> User-Agent: curl/7.29.0
> Host: icinga-test
> Accept: application/json
>
< HTTP/1.1 401 Unauthorized
HTTP/1.1 401 Unauthorized
< Date: Fri, 22 Dec 2017 12:51:54 GMT
Date: Fri, 22 Dec 2017 12:51:54 GMT
< Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_wsgi/3.4 Python/2.7.5
Server: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_wsgi/3.4 Python/2.7.5
< X-Powered-By: PHP/7.1.8
X-Powered-By: PHP/7.1.8
* Authentication problem. Ignoring this.
< Www-Authenticate: Basic realm="Icinga Web 2"
Www-Authenticate: Basic realm="Icinga Web 2"
< Strict-Transport-Security: max-age=15768000;includeSubdomains
Strict-Transport-Security: max-age=15768000;includeSubdomains
< Transfer-Encoding: chunked
Transfer-Encoding: chunked
< Content-Type: text/html; charset=UTF-8
Content-Type: text/html; charset=UTF-8

<
* Connection #0 to host icinga-test left intact
```

## Expected Behavior

Authentication using HTTP Basic Auth with Icingaweb2 user accounts should work according to the documentation.

## Current Behavior

Sadly, it doesn't work for me.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Your Environment

* Director version (System - About): Git from today
* Icinga Web 2 version and modules (System - About): 2.5.0
* Icinga 2 version (`icinga2 --version`): 2.8.0-r1
* Operating System and version: CentOS 7.4 x64
* Webserver, PHP versions: Apache 2.4, PHP 7.1 from SCL repo

**Comment by mkayontour** [comment]
I'm facing a similar issue - i can't even get the basic requests to work. 
```
curl -k -u 'admin:admin' -H 'Accept: application/json' 'https://xx.xx.xx.50/icingaweb2/director/host?name=hostnamep11' --trace-ascii - -X PROPFIND -D -
== Info: About to connect() to xx.xx.xx.50 port 443 (#0)
== Info:   Trying xx.xx.xx.50...
== Info: Connected to xx.xx.xx.50 (xx.xx.xx.50) port 443 (#0)
== Info: Initializing NSS with certpath: sql:/etc/pki/nssdb
== Info: skipping SSL peer certificate verification
== Info: SSL connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
== Info: Server certificate:
== Info:        subject: CN=*.customer.de,O=customer AG,L=Hamburg,ST=Hamburg,C=DE
== Info:        start date: Aug 17 00:00:00 2015 GMT
== Info:        expire date: Apr 16 23:59:59 2018 GMT
== Info:        common name: *.customer.de
== Info:        issuer: CN=GeoTrust SSL CA - G3,O=GeoTrust Inc.,C=US
== Info: Server auth using Basic with user 'admin'
=> Send header, 175 bytes (0xaf)
0000: PROPFIND /icingaweb2/director/host?name=hostnamep11 HTTP/1.1
003e: Authorization: Basic YWRtaW46YWRtaW4=
0065: User-Agent: curl/7.29.0
007e: Host: xx.xx.xx.50
0093: Accept: application/json
00ad:
<= Recv header, 27 bytes (0x1b)
0000: HTTP/1.1 401 Unauthorized
HTTP/1.1 401 Unauthorized
<= Recv header, 37 bytes (0x25)
0000: Date: Thu, 04 Jan 2018 09:15:01 GMT
Date: Thu, 04 Jan 2018 09:15:01 GMT
<= Recv header, 16 bytes (0x10)
0000: Server: Apache
Server: Apache
<= Recv header, 25 bytes (0x19)
0000: X-Powered-By: PHP/7.1.8
X-Powered-By: PHP/7.1.8
== Info: Authentication problem. Ignoring this.
<= Recv header, 46 bytes (0x2e)
0000: Www-Authenticate: Basic realm="Icinga Web 2"
Www-Authenticate: Basic realm="Icinga Web 2"
<= Recv header, 28 bytes (0x1c)
0000: Transfer-Encoding: chunked
Transfer-Encoding: chunked
<= Recv header, 40 bytes (0x28)
0000: Content-Type: text/html; charset=UTF-8
Content-Type: text/html; charset=UTF-8

<= Recv header, 2 bytes (0x2)
0000:
<= Recv data, 5 bytes (0x5)
0000: 0
0003:
== Info: Connection #0 to host xx.xx.xx.50 left intact
```

Same Environment as @terra-nova

**Comment by Thomas-Gelf** [comment]
This has been broken with Icinga Web 2. The issue duplicates [Icinga Web 2 issue #3245](https://github.com/Icinga/icingaweb2/issues/3245) and that one duplicates [#3173](https://github.com/Icinga/icingaweb2/issues/3173). The related fix is [644cf0f1](https://github.com/Icinga/icingaweb2/commit/644cf0f1c38ebba899b3ab5ab97a85b2a0d9e469) in Icinga Web 2, you can apply it to your web server configuration.

Cheers,
Thomas

**Comment by mmahfoudi** [comment]
This error is still present...
